### PR TITLE
feat: added pages

### PR DESCRIPTION
**Pull request by mpehlivanova**

### CODE REVIEW

File: src/auth/auth-manager.ts [hunk 1]
```diff
@@ -47,10 +47,11 @@ export const AuthManager = () => {
   };
 
   const isValidIdToken: () => boolean = () => {
-    const { idTokenClaims }: any = msalInst?.getActiveAccount() || null;
+    const accessToken: string = getAccessToken() || '';
```
Comment (igargov) [inline src/auth/auth-manager.ts:51]: You have changed the names of the variables to `accessToken` but still the method name is `isValidIdToken`. We have to align the name of the variables and the method itself.

File: src/enum.ts [hunk 2]
```diff
@@ -7,6 +7,22 @@ export enum Pages {
...
+  video = 'video',
+}
+
+export enum PERMISSIONS {
+  DELETE = 'delete:users',
+  READ = 'read:users',
+  WRITE = 'write:users',
+  CREATE = 'create:users',
+  EDIT = 'edit:users',
+  UPDATE = 'update:user',
```
Comment (igargov) [inline src/enum.ts]: The scope of the permission differs from the the scope of the other permissions. `user` in this case. Have in mind that Write, Edit and Update usually are used interchangeably. Maybe it is a good idea to choose one for edit operations.

File: src/layout/constants.tsx [hunk 3]
```diff
@@ -1,37 +1,116 @@
...
-export const routeConfiguration: RouteConfigurationType[] = [
-  {
-    id: Pages.login,
-    isRequireAuth: false,
-    url: '/',
-    component: <LoginPage />,
+
+export const groupPermission = {
+  READ_WRITE: [PERMISSIONS.READ, PERMISSIONS.WRITE],
+  EDIT_UPDATE: [PERMISSIONS.EDIT, PERMISSIONS.UPDATE],
```
Comment (igargov) [inline src/layout/constants.tsx]: Here we can apply the same idea- edit and update are practically the same thing.

File: src/auth/restricted-access.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+export const RestrictedAccess: React.FC<{ children: JSX.Element, permissionsPage?: PERMISSIONS[], permissionsUser: PERMISSIONS[] }> = ({
+  permissionsPage, permissionsUser, children
+}) => {
+  const hasPermission: boolean | undefined = permissionsPage?.reduce((acc: boolean, current: PERMISSIONS) => {
+    return acc && Boolean(permissionsUser?.includes(current));
+  }, true);
+  if (hasPermission) {
+    return children;
+  }
+  return null;
```
Comment (igargov) [inline src/auth/restricted-access.tsx]: I don't think that it is good to return null directly. This means that we would need to have explicit null check in the preceding component. I would personally reduce the use of "return null" statements as much as possible.

Comment (mpehlivanova) [reply src/auth/restricted-access.tsx]: fixed

File: src/layout/constants.tsx (hunk 3 above)
Comment (mpehlivanova) [reply src/layout/constants.tsx]: fixed

File: src/enum.ts (hunk 2 above)
Comment (mpehlivanova) [reply src/enum.ts]: fixed

File: src/auth/auth-manager.ts (hunk 1 above)
Comment (mpehlivanova) [reply src/auth/auth-manager.ts:51]: fixed

**Review by igargov — APPROVED** [review]

**Pull request merged**